packages/x-tree-view/src/internals/useTreeView/useTreeView.ts

```diff
@@ -148,23 +149,26 @@ export const useTreeView = <Plugins extends readonly TreeViewPlugin<TreeViewAnyP
 
   const itemWrappers = plugins
     .map((plugin) => plugin.wrapItem)
-    .filter((wrapItem): wrapItem is TreeItemWrapper => !!wrapItem);
+    .filter((wrapItem): wrapItem is TreeItemWrapper<any> => !!wrapItem);
   contextValue.wrapItem = ({ itemId, children }) => {
     let finalChildren: React.ReactNode = children;
     itemWrappers.forEach((itemWrapper) => {
-      finalChildren = itemWrapper({ itemId, children: finalChildren });
+      finalChildren = itemWrapper({ itemId, children: finalChildren, instance });
     });
 
     return finalChildren;
   };
 
   const rootWrappers = plugins
     .map((plugin) => plugin.wrapRoot)
-    .filter((wrapRoot): wrapRoot is TreeRootWrapper => !!wrapRoot);
+    .filter((wrapRoot): wrapRoot is TreeRootWrapper<any> => !!wrapRoot)
+    // The wrappers are reversed to ensure that the first wrapper is the outermost one.
+    .reverse();
```

Comment:
Otherwise the provider of `useTreeViewItems` takes precedence over `useTreeVIewJSXItems` which messes the behavior on `SimpleTreeView`

IMHO it make sense that if two plugins wrap the root (or the items), then it's the last executed plugin that has its wrappers deeper in the component tree and thus executed last.